addons/sale/models/sale_order.py

```diff
@@ -634,7 +639,13 @@ def _compute_invoice_status(self):
                 ['order_id', 'invoice_status']
             )
         ]
-        for order in confirmed_orders:
+
+        manually_closed_orders = self.filtered(
+            lambda o: o.is_manually_closed and o.state == 'sale'
+        )
+        manually_closed_orders.invoice_status = 'invoiced'
+        orders_to_compute = confirmed_orders - manually_closed_orders
```

Comment:
move this right after confirm_orders filter as we don't need to check `line_invoice_status_all` for manually closed orders